homeassistant/components/ledsc/config_flow.py

```diff
@@ -0,0 +1,68 @@
+"""Config flow for Hello World integration."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_PORT
+from homeassistant.core import HomeAssistant
+
+from .consts import DEFAULT_HOST, DEFAULT_PORT, DOMAIN
+from websc_client import WebSClientAsync as WebSClient
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST, default=DEFAULT_HOST): str,
+        vol.Required(CONF_PORT, default=DEFAULT_PORT): int,
+    }
+)
+
+
+async def validate_input(hass: HomeAssistant, data: dict) -> dict[str, Any]:
+    """
+    Validate the user input allows us to connect.
+
+    Data has the keys from DATA_SCHEMA with values provided by the user.
+    """
+    host = data[CONF_HOST]
+    port = data[CONF_PORT]
+
+    client = WebSClient(host, port)
+    await client.connect()
+    await client.disconnect()
+
+    return {"title": f"LedSC server {host}:{port}"}
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Hello World."""
+
+    VERSION = 1
+
+    async def async_step_user(self, user_input: dict[str, Any] | None = None):
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+        if user_input:
+            self._async_abort_entries_match(
+                {
+                    CONF_HOST: user_input[CONF_HOST],
+                    CONF_PORT: user_input[CONF_PORT],
+                }
+            )
+
+            try:
+                info = await validate_input(self.hass, user_input)
+            except ConnectionError:
+                errors["base"] = "Cannot connect"
```

Comment:
This is not the error message that is displayed, it is a tranlation key and the corresponding message is looked up in the `strings.json` file. Please add a `strings.json` file to your integration.
```suggestion
                errors["base"] = "cannot_connect
```

Reply:
Done